ouroboros-consensus/src/Ouroboros/Consensus/Storage/ImmutableDB/Impl/Index/Primary.hs

```diff
@@ -17,7 +17,6 @@ module Ouroboros.Consensus.Storage.ImmutableDB.Impl.Index.Primary (
     SecondaryOffset
     -- * PrimaryIndex
   , PrimaryIndex (..)
-    -- Exported for the benefit of tests
```

Comment:
I now see the

> For those types opaque comment was removed completely.

sentence in your commit message.

We should retain the `opaque` or `for testing` annotations in some way; I don't want to just drop them.

As you noticed -- I'm catching on now -- there are pretty often pre-existing sections where we could just relocate the `opaque` comment up into the section header. So: `-- * PrimaryIndex (opaque)`.

Reply:
I've addressed all opaque types in latest commit (https://github.com/input-output-hk/ouroboros-network/pull/3009/commits/e0d2510add8fc7a5c6f578fa38ac5f1e93bd47fc)